cmd/api/src/api/registration/v2.go

```diff
@@ -51,8 +45,10 @@ func registerV2Auth(cfg config.Configuration, db database.Database, permissions
 		routerInst.GET("/api/v2/self", managementResource.GetSelf),
 		routerInst.POST("/api/v2/logout", loginResource.Logout),
 
-		// Login path prefix matcher for SAML providers
-		routerInst.PathPrefix("/api/v2/login/saml/{saml_provider_name}", middleware.ContextMiddleware(samlResource)),
+		// Login path prefix matcher for SAML providers, order matters here due to PathPrefix
+		routerInst.POST("/api/{version}/login/saml/{saml_provider_name}/acs", managementResource.SAMLCallbackRedirect),
+		routerInst.GET("/api/{version}/login/saml/{saml_provider_name}/metadata", managementResource.ServeMetadata),
+		routerInst.PathPrefix("/api/{version}/login/saml/{saml_provider_name}", http.HandlerFunc(managementResource.SAMLLoginRedirect)),
```

Comment:
Forced to do path prefix here because some use `/sso` as the entrypoint and others use root path